plugin/interestingwords.vim

```diff
@@ -197,29 +197,27 @@ function! s:buildColors()
   if (s:hasBuiltColors)
     return
   endif
-  if has('gui_running')
-    let ui = 'gui'
-    let wordColors = g:interestingWordsGUIColors
-  else
-    let ui = 'cterm'
-    let wordColors = g:interestingWordsTermColors
-  endif
+  let nGUIColors = len(g:interestingWordsGUIColors)
+  let nTermColors = len(g:interestingWordsTermColors)
+  let colorIndices = range(max([nGUIColors, nTermColors]))
```

Comment:
Correct, I was trying to warn about this in the first post. The number of available visual groups will always be _max(len(g:interestingWordsGUIColors), len(g:interestingWordsTermColors))_, and it can lead to an even more confusing problem than the one you pointed out: if user defines say only 3 term colors (and no gui colors), he will have 6 groups to cycle through with each color repeating itself, since 6 gui colors are defined by the plugin itself.

Unfortunately, [short of using this monstrosity](https://github.com/vim-airline/vim-airline/blob/master/autoload/airline/init.vim#L113-L115) which properly checks which colors should be used, I don't think there is a nice fix for this.

Could you please give me a use case for that feature though?